play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/cache/CacheableResponse.scala

```diff
@@ -200,17 +197,20 @@ case class CacheableResponse(
   }
 
   private def buildCookies: util.List[Cookie] = {
+    import play.shaded.ahc.io.netty.handler.codec.http.cookie.ClientCookieDecoder
     import play.shaded.ahc.org.asynchttpclient.util.MiscUtils.isNonEmpty
-    import play.shaded.ahc.org.asynchttpclient.cookie.CookieDecoder
     import java.util.Collections
 
-    var setCookieHeaders = headers.getHeaders.getAll(SET_COOKIE2)
-    if (!isNonEmpty(setCookieHeaders)) setCookieHeaders = headers.getHeaders.getAll(SET_COOKIE)
+    var setCookieHeaders = headers.getAll(SET_COOKIE2)
+    if (!isNonEmpty(setCookieHeaders)) setCookieHeaders = headers.getAll(SET_COOKIE)
     if (isNonEmpty(setCookieHeaders)) {
       val cookies = new util.ArrayList[Cookie](1)
       import scala.collection.JavaConversions._
       for (value <- setCookieHeaders) {
-        val c = CookieDecoder.decode(value)
+        val c = ahcConfig.isUseLaxCookieEncoder match {
```

Comment:
This logic is duplicated. It could be good to extract it to a method that can be reused.